controllers/storagecluster/server.go

```diff
@@ -0,0 +1,190 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/multierr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+)
+
+var (
+	ServerImage = "quay.io/sp1098/ocs-operator:consumer-api"
+)
+
+type ocsProviderServer struct{}
+
+func (obj *ocsProviderServer) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	if !instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("nigoyal Spec.AllowRemoteStorageConsumers is disabled")
+		return obj.ensureDeleted(r, instance)
+	}
+
+	r.Log.Info("nigoyal Spec.AllowRemoteStorageConsumers is enabled creating resources")
+
+	desiredServerService := GetProviderServerService(instance)
+	desiredServerDeployment := GetProviderServerDeployment(instance)
+
+	var finalErr error
+
+	for _, obj := range []client.Object{
```

Comment:
...don't use `obj` here, this is extremely confusing when the *struct you are writing a method for* is also `obj`. :stuck_out_tongue_closed_eyes: Honestly, though, you should change all the instances of `(obj *ocsProviderServer)` to something like `(srv *ocsProviderServer)`.

Reply:
changed for loop obj to `resource` and method `obj` to `o`